### PR TITLE
Add support of redis clusters

### PR DESCRIPTION
**Pull request by Millefeuille42**
Fixing https://github.com/bbangert/beaker/issues/235

### CODE REVIEW

File: beaker/ext/redisclusternm.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    The data will be stored into redis keys, with their name
+    starting with ``beaker_cache:``. So make sure you provide
+    a specific database number if you don't want to mix them
+    with your own data.
+    """
+    MAX_KEY_LENGTH = 1024
+
+    clients = SyncDict()
+
+    def __init__(self, namespace, url, timeout=None, **kw):
```
Comment (lebouquetin) [inline beaker/ext/redisclusternm.py]: In `__init__` signature and code:
- Rename url to urls

In the class docstring:
- add explaination that that urls must be comma separated + gives an example, something like `urls: set list of redis cluster urls separated by a comma. Example: https://192.168.1.2:6679,https://192.168.1.3:6679,https://192.168.1.4:6679`
- add explanation about not putting a db name in the url

File: beaker/ext/redisclusternm.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    an already existing RedisCluster instance.
+
+    This Synchronizer only supports 1 reader or 1 writer at time, not concurrent readers.
+    """
+    # If a cache entry generation function can take a lot,
+    # but 15 minutes is more than a reasonable time.
+    LOCK_EXPIRATION = 900
+    MACHINE_ID = machine_identifier()
+
+    def __init__(self, identifier, url):
```
Comment (lebouquetin) [inline beaker/ext/redisclusternm.py]: is it one url or several urls? If can be a list of urls, then add similar comment like for  RedisClusterNamespaceManager

File: beaker/ext/redisclusternm.py (hunk 1 above)
Comment (Millefeuille42) [reply beaker/ext/redisclusternm.py]: fixed.

File: beaker/ext/redisclusternm.py (hunk 2 above)
Comment (Millefeuille42) [reply beaker/ext/redisclusternm.py]: It is several urls, applied same suggestion as above.

**Comment by lebouquetin** [comment]
Add Redis cluster support as session management backend

**Pull request merged**